activesupport/lib/active_support/core_ext/string/access.rb

```diff
@@ -61,7 +61,8 @@ def from(position)
   #   str.from(0).to(-1) # => "hello"
   #   str.from(1).to(-2) # => "ell"
   def to(position)
-    self[0..position]
+    position = [position + length, -1].max if position < 0
```

Comment:
Curious what the -1 and `max` here is for? I could get all tests to pass with: `position += size if position < 0`.

Reply:
I suppose the condition does not preserve the original behavior, or hard to understand what this do.
Maybe we should more exercise that in tests.

Reply:
It is for when `position + length` is less than `-1`.

Reply:
@rafaelfranca  Yes, exactly.

For example, `"foo".to(-10)` without the `max` becomes `"foo"[0, -10 + 3 + 1]` or `"foo"[0, -6]`, which returns `nil`.

Reply:
Ah, I see! I found relying on -1 + 1 to get a 0 length for String#[] a little confusing, so I tried to change it a bit in efa2299ac3 with extra tests.

Reply:
@kaspth I had considered that variation, because it is a bit more clear.  But, I decided to use `max` because the extra `to_s` makes it ~13% slower when the position isn't negative.  Of course, it's only a micro-benchmark and likely of no consequence, but since the purpose of the PR was performance, that's what I went with.

Reply:
Changed to fastest variant 71e41a5b9495ba6ebe3cdf1e10afa27bcc0db208.

Reply:
@kamipo Excellent!  :+1: